### PR TITLE
Basic rest api

### PR DESCRIPTION
**Pull request by roznawsk**

### CODE REVIEW

File: generate_client.sh
Comment (sgfn) [inline generate_client.sh:1]: Make sure to `chmod +x` this file and add `#!/bin/bash` as the first line

File: generate_client.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+rm -rf openapi
+openapi-generator-cli generate \
+    -i https://raw.githubusercontent.com/jellyfish-dev/jellyfish/openapi-generator-compatibility/openapi.yaml \
```
Comment (sgfn) [inline generate_client.sh:3]: Don't forget to revert to `main` once everything gets merged

File: generate_client.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    -g python \
+    -o openapi \
+    --package-name openapi_client \
+    --global-property apis,models,modelTests=false,supportingFiles
+    
+rm -rf openapi/{docs,test,.github,.openapi-generator,.gitignore,.gitlab-ci.yml,.travis.yml,.openapi-generator-ignore,git_push.sh,README.md,setup.cfg,test-requirements.txt,tox.ini}
```
Comment (sgfn) [inline generate_client.sh:9]: Just to confirm, are you sure we want to delete `.gitignore`?

File: openapi/openapi_client/__init__.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# coding: utf-8
+
+# flake8: noqa
+
+"""
+    Jellyfish Media Server
+
+    No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
```
Comment (sgfn) [inline openapi/openapi_client/__init__.py:8]: If it's an easy fix, you could copy over the basic Jellyfish description (but that's not a big issue)

File: jellyfish/room_api.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        self._configuration = jellyfish_api.Configuration(
+            host=server_address,
+            access_token=server_api_token
+        )
+
+        self._api_client = jellyfish_api.ApiClient(self._configuration)
+        self._room_api = jellyfish_api.RoomApi(self._api_client)
+
+    def create_room(self, max_peers: int = None, video_codec: str = None) -> (str, Room):
+        """Creates a room"""
```
Comment (sgfn) [inline jellyfish/room_api.py:24]: I don't know where we're planning to put the documentation (when everything's finished), but I think there should be some explanation of the return values in the docstring here

Now that I look at it, there should really be something like that in our elixir sdk as well...

File: jellyfish/room_api.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        """Returns list of all rooms"""
+
+        return self._room_api.get_all_rooms().data
+
+    def get_room(self, room_id: str) -> Room:
+        """Returns room with the given id"""
+
+        return self._room_api.get_room(room_id).data
+
+    def add_peer(self, room_id: str, peer_type: str, options) -> (str, Peer):
```
Comment (sgfn) [inline jellyfish/room_api.py:46]: `options` aren't typed?

File: jellyfish/room_api.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    def delete_peer(self, room_id: str, peer_id: str) -> None:
+        """Deletes peer"""
+
+        return self._room_api.delete_peer(room_id, peer_id)
+
+    def add_component(self, room_id: str, component_type: str, options=None) -> Component:
+        """Creates component in the room"""
+
+        if options:
+            options = ComponentOptions(options)
```
Comment (sgfn) [inline jellyfish/room_api.py:64]: No idea if what I'm about to suggest is consistent with the style guide/PEP whichever, but I fear things like `0`, `[]`, `{}` etc. might evaluate to `false` and break something later.

```suggestion
        if options is not None:
            options = ComponentOptions(options)
```

In this case, it's probably safe though, so you can ignore it

File: openapitools.json [hunk 7]
```diff
@@ -0,0 +1,7 @@
+{
+  "$schema": "./node_modules/@openapitools/openapi-generator-cli/config.schema.json",
```
Comment (sgfn) [inline openapitools.json:2]: Where is this?

**Review by sgfn — COMMENTED** [review]
Nice codec in tests

File: generate_client.sh (hunk 2 above)
Comment (roznawsk) [reply generate_client.sh:9]: Yes, but just the generated one inside `openapi/`

File: jellyfish/room_api.py (hunk 4 above)
Comment (roznawsk) [reply jellyfish/room_api.py:24]: Yeah I thought about it as well. I decided to leave it for later though. We have to find a tool for generating docs from docstrings. We may also decide to use some additional typing libraries.

File: jellyfish/room_api.py (hunk 5 above)
Comment (roznawsk) [reply jellyfish/room_api.py:46]: So the only actual value we can put here is `ComponentOptionsRTSP` or `dict()` (empty options for HLS component). However with default python typing you can't specify alternative so I decided to leave this one without typing.

File: openapitools.json (hunk 7 above)
Comment (roznawsk) [reply openapitools.json:2]: Good question, I don't know but this is autogenerated. Personally, the important part of this file is the generator version.

File: openapi/openapi_client/__init__.py (hunk 3 above)
Comment (roznawsk) [reply openapi/openapi_client/__init__.py]: done!

**Review by sgfn — APPROVED** [review]
🥇

**Pull request merged**